candlelight.go

```diff
@@ -16,8 +16,81 @@
  */
 package candlelight
 
-import ()
+import (
+	"github.com/go-kit/kit/log"
+	"github.com/spf13/viper"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/stdout"
+	"go.opentelemetry.io/otel/exporters/trace/jaeger"
+	"go.opentelemetry.io/otel/exporters/trace/zipkin"
+	"go.opentelemetry.io/otel/label"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
 
-func HelloWorld() error {
-	return nil
+const (
+	zipkinName                   = "zipkin"
+	jaegarName                   = "jaegar"
+	traceProviderType            = "type"
+	traceProviderEndpoint        = "endpoint"
+	traceProviderSkipTraceExport = "skipTraceExport"
+)
+
+/***
+ 1. will be responsible for creating the traceprovider and setting it back to  opentelemetry
+		viper will be having all the fields like  type,endpoint and skipTraceExport
+ 2. supported traceProviders are zipkin,jaegar and stdout
+ 3. set skipTraceExport = true if you don't want to print the span and tracer information in stdout
+*/
+func ConfigureTracerProvider(v *viper.Viper, logger log.Logger, applicationName string) {
```

Comment:
Should the configuration return an error or nil to indicate if there was an issue?

Reply:
we can't return nil if the configuration is missing in properties.  if we are doing this then we will have issues when we are starting span. I had written the code in such a way if nothing is specified it will be taking the default as stdout and will be creating the global tracer based on that.

example usage: candlelight.ConfigureTracerProvider(v.Sub("traceProvider"),logger,applicationName)
if traceProvider is missing in the properties file then viper will be coming as nil in this case we are creating the stdout tracer.

Yes, we can return an error that will not start the application.


What do you suggest should we continue like this or we should return an error which will not start the application?

Reply:
Of the two I'd return an error and not start the application to help prevent a silent failure that could be more costly (detected when you need it at runtime).

Reply:
I had updated the code as per this conversation. now we need to handle the error where we are using this module.